OpenRA.Game/Game.cs

```diff
@@ -457,6 +465,28 @@ static string ChooseShellmap()
 			return shellmaps.Random(CosmeticRandom);
 		}
 
+		public static void SwitchToExternalMod(ExternalMod mod, string[] launchArguments = null, Action onFailed = null)
+		{
+			try
+			{
+				var argsString = mod.LaunchArgs.Append(launchArguments)
+					.Select(a => "\"" + a + "\"").JoinWith(" ");
+
+				var p = Process.Start(mod.LaunchPath, argsString);
```

Comment:
> I've set this up in a way that will let us drop in "remote" external mod data in the same way that we do for maps. At some point in the future i'd like to think that we would add a mod registration form to the resource center, so that we can show the icon/title/author/website of third party mods even if they aren't installed.